src/parser/func_type.cpp

```diff
@@ -0,0 +1,155 @@
+#include "func_type.hpp"
+#include "parse_tree.h"
+#include "parser.h"
+
+std::vector<func_signature> func_signature::integer = {
+  {PT_TYPE_INTEGER, {}, {}},
+};
+
+std::vector<func_signature> func_signature::bigint = {
+  {PT_TYPE_BIGINT, {}, {}},
+};
+
+std::vector<func_signature> func_signature::double_double01 = {
+  {PT_TYPE_DOUBLE, {PT_TYPE_DOUBLE}, {}},
+};
+
+std::vector<func_signature> func_signature::percentile_dis = {
+  {PT_TYPE_MAYBE, {PT_GENERIC_TYPE_ANY}, {}},
+};
+
+std::vector<func_signature> func_signature::bigint_discrete = {
```

Comment:
If I put it as a comment and the association functionID-signature (from func_type.x) will change, then it will need to be maintained or it will become inaccurate and because I cannot enforce somehow the update of the comment, I'll prefer not to do it at all.
As I said, the association is maintained in one place: func_type.x (X-macro style)